dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -70,296 +69,365 @@
+    private final String suiteFileName;
+    private final Set<String> versionedJars;
+    private final Map<String, String> additionalMvnProps;
+
+    /**
+     * runs "mvn clean test" in the tck folder
+     *
+     * @param server the liberty server which should be used to run the TCK
+     * @param bucketName the name of the test project
+     * @param testName the name of the method that's being used to launch the TCK
      */
-    public static void setSuiteFileName(String newName, LibertyServer server) throws Exception {
-        overrideSuiteFileName = newName;
-        mvnOutputFilename = "mvnOutput_" + newName.replace(".xml", "");
-        targetFolder = defaultTargetFolder + "_" + newName.replace(".xml", "");
+    public static int runTCKMvnCmd(LibertyServer server, String bucketName, String testName) throws Exception {
+        return runTCKMvnCmd(server, bucketName, testName, DEFAULT_SUITE_FILENAME, Collections.<String, String> emptyMap(), Collections.<String> emptySet());
     }
 
     /**
-     * Initialise shared values for a particular server.
-     * This enables us to set up things like the Liberty install
-     * directory and jar locations once.
+     * runs "mvn clean test" in the tck folder, passing through all the required properties
      *
-     * @param server Simplicity LibertyServer
-     * @throws Exception
+     * @param server the liberty server which should be used to run the TCK
+     * @param bucketName the name of the test project
+     * @param testName the name of the method that's being used to launch the TCK
+     * @param additionalProps java properties to set when running the mvn command
      */
-    private static void init(LibertyServer server) throws Exception {
-        String pomRelativePath = "tck/pom.xml";
-        wlp = server.getInstallRoot();
-        resultsDir = Props.getInstance().getFileProperty(Props.DIR_LOG); //typically ${component_Root_Directory}/results
-        componentRootDir = Props.getInstance().getFileProperty(Props.DIR_COMPONENT_ROOT);
-
-        tckRunnerDir = new File("publish/tckRunner");
-
-        pomXml = new File(tckRunnerDir, pomRelativePath);
+    public static int runTCKMvnCmd(LibertyServer server, String bucketName, String testName, Map<String, String> additionalProps) throws Exception {
+        return runTCKMvnCmd(server, bucketName, testName, DEFAULT_SUITE_FILENAME, additionalProps, Collections.<String> emptySet());
+    }
 
-        if (!init) {
-            populateJarsFromWlp(pomXml);
-        }
+    /**
+     * runs "mvn clean test" in the tck folder, passing through all the required properties
+     *
+     * @param server the liberty server which should be used to run the TCK
+     * @param bucketName the name of the test project
+     * @param testName the name of the method that's being used to launch the TCK
+     * @param suiteFileName the name of the suite xml file
+     * @param additionalProps java properties to set when running the mvn command
```

Comment:
Are "java properties" system properties?

Reply:
yes ... they get turned into `-Dkey=value` elements in the mvn command line